cg/store/crud/read.py

```diff
@@ -456,6 +444,49 @@ def get_sample_lane_sequencing_metrics_by_flow_cell_name(
             flow_cell_name=flow_cell_name,
         ).all()
 
+    def get_illumina_metrics_entry_by_device_sample_and_lane(
+        self, run_id: str, sample_internal_id: str, lane: int
+    ) -> IlluminaSampleSequencingMetrics:
+        """Get metrics entry by sequencing device internal id, sample internal id and lane."""
+        filtered_samples: Subquery = apply_sample_filter(
+            samples=self._get_query(table=Sample),
+            filter_functions=[SampleFilter.BY_INTERNAL_ID],
+            internal_id=sample_internal_id,
+        ).subquery()
+        filtered_metrics: Subquery = apply_illumina_metrics_filter(
+            metrics=self._get_query(table=IlluminaSampleSequencingMetrics),
+            filter_functions=[IlluminaMetricsFilter.BY_LANE],
+            lane=lane,
+        ).subquery()
+        filtered_flow_cells: Subquery = apply_illumina_flow_cell_filters(
+            flow_cells=self._get_query(table=IlluminaFlowCell),
+            filter_functions=[IlluminaFlowCellFilter.BY_INTERNAL_ID],
+            internal_id=run_id,
+        ).subquery()
+
+        joined_query: Query = (
+            self._get_query(table=IlluminaSampleSequencingMetrics)
+            .select_from(IlluminaSampleSequencingMetrics)
+            .join(
+                filtered_samples, IlluminaSampleSequencingMetrics.sample_id == filtered_samples.c.id
+            )
+            .join(filtered_metrics, IlluminaSampleSequencingMetrics.id == filtered_metrics.c.id)
+            .join(
+                IlluminaSequencingRun,
+                IlluminaSampleSequencingMetrics.instrument_run_id == IlluminaSequencingRun.id,
+            )
+            .join(filtered_flow_cells, IlluminaSequencingRun.device_id == filtered_flow_cells.c.id)
+        )
```

Comment:
Can we extract the join logic for this into a `_join_illumina_tables` that performs the joins between the `IlluminaFlowCell`, `IlluminaSequencingRun` and `IlluminaSampleSequencingMetrics` I think this join always be the same anyway. 

Then after that join with the samples?